tests/test_debugging.py

```diff
@@ -14,121 +14,320 @@
 """
 Unit tests for the debugging module.
 """
+from contextlib import nullcontext
 from unittest.mock import patch
 
 import numpy as np
 import pytest
+from flaky import flaky
 
 import pennylane as qml
 from pennylane import numpy as qnp
 from pennylane.debugging import PLDB, pldb_device_manager
 
 
-class TestSnapshot:
+def _compare_numpy_dicts(dict1, dict2):
+    assert dict1.keys() == dict2.keys()
+    assert all(np.allclose(dict1[key], dict_2_val) for key, dict_2_val in dict2.items())
+
+
+class TestSnapshotTape:
+    def test_snapshot_output_tapes(self):
+        ops = [
+            qml.Snapshot(),
+            qml.Hadamard(wires=0),
+            qml.Snapshot("very_important_state"),
+            qml.CNOT(wires=[0, 1]),
+            qml.Snapshot(),
+        ]
+
+        measurements = [qml.expval(qml.PauliX(0))]
+
+        num_snapshots = len(tuple(filter(lambda x: isinstance(x, qml.Snapshot), ops)))
+
+        tape = qml.tape.QuantumTape(ops, measurements)
+
+        tapes, _ = qml.snapshots(tape)
+
+        assert len(tapes) == num_snapshots + 1
+
+        tape_no_meas = qml.tape.QuantumTape(ops)
+
+        tapes_no_meas, _ = qml.snapshots(tape_no_meas)
+
+        assert len(tapes_no_meas) == num_snapshots
```

Comment:
I'd prefer to add in a few more explicit tests for the actual content of the batches. 

```suggestion
        assert len(tapes_no_meas) == num_snapshots
        tape1 = qml.tape.QuantumScript([], [qml.state()])
        assert qml.equal(tapes_no_meas[0], tape1)
        tape2 = qml.tape.QuantumScript([qml.Hadamard(0)], [qml.state()])
        assert qml.equal(tapes_no_meas[1], tape2)
        tape3 = qml.tape.QuantumScript([qml.Hadamad(0), qml.CNOT((0,1))], [qml.state()])
        assert qml.equal(tapes_no_meas[2], tape3)
        tape4 = qml.tape.QuantumScript([qml.Hadamard(0), qml.CNOT((0,1))], [qml.expval(qml.X(0))])
        assert qml.equal(tapes_no_meas[3], tape4)
```

I would also recommend having some of the snapshots here include measurements.

A lot of things could be happening inside `qml.snapshot.transform` that still give rise to the correct length of batch.